ldi-core/version-materialiser/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/VersionMaterialiser.java

```diff
@@ -22,36 +20,39 @@ public VersionMaterialiser(Property versionPredicate, boolean restrictToMembers)
 
 	@Override
 	public Model apply(Model linkedDataModel) {
-		Model versionMaterialisedModel = ModelFactory.createDefaultModel();
+		final Model versionMaterialisedModel = ModelFactory.createDefaultModel();
 
 		Map<Resource, Resource> versionIDEntityIDMap = getVersionIDEntityIDMap(linkedDataModel, versionPredicate);
 
-		linkedDataModel.listStatements().forEach(statement -> {
-			Resource subject = statement.getSubject();
-			Property predicate = statement.getPredicate();
-			RDFNode object = statement.getObject();
-
-			// Statement needs 'de-versioning', replacing the subject.
-			if (versionIDEntityIDMap.containsKey(statement.getSubject()))
-				subject = versionIDEntityIDMap.get(statement.getSubject());
-
-			// Object references a versioned entity, replace it with the 'de-versioned'
-			// identifier.
-			if (statement.getObject().isResource()
-					&& versionIDEntityIDMap.containsKey(statement.getObject()))
-				object = versionIDEntityIDMap.get(statement.getObject());
+		List<Statement> deversionedStatements = linkedDataModel.listStatements()
+				.toList().stream()
+				.filter(statement -> !statement.getPredicate().equals(versionPredicate)) // drop isVersionOf stmts
+				.map(statement -> deversionStatement(versionIDEntityIDMap, statement))
+				.toList();
 
-			// Don't add isVersionOf statements.
-			if (statement.getPredicate().equals(versionPredicate))
-				return;
+		versionMaterialisedModel.add(deversionedStatements);
 
-			versionMaterialisedModel.add(new StatementImpl(subject, predicate, object));
-		});
-		Model finalVersionMaterialisedModel = versionMaterialisedModel;
 		if (restrictToMembers) {
-			finalVersionMaterialisedModel = reduceToLDESMemberOnlyModel(finalVersionMaterialisedModel);
+			return reduceToLDESMemberOnlyModel(versionMaterialisedModel);
 		}
-		return finalVersionMaterialisedModel;
+		return versionMaterialisedModel;
+	}
+
+	private Statement deversionStatement(Map<Resource, Resource> versionIdEntityIdMap, Statement statement) {
+		Resource subject = statement.getSubject();
+		RDFNode object = statement.getObject();
+
+		// Statement needs 'de-versioning', replacing the subject.
+		if (versionIdEntityIdMap.containsKey(subject))
```

Comment:
can you add curly braces? We do this to prevent :) the infamous https://www.wired.com/2014/02/gotofail/